### PR TITLE
Improvement/Bring docs into monorepo

### PR DESCRIPTION
**Pull request by dabeeeenster**
This PR will bring in the docs to the main repo for many benefits!

Once the PR is merged we can set up the vercel integration to auto build docs.flagsmith.com.

Some notes:

- Removed husky pre-commit as the linting is now managed by the mono-repo with https://pre-commit.com/

### CODE REVIEW

File: docs/docusaurus.config.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    presets: [
+        [
+            'docusaurus-preset-openapi',
+            {
+                docs: {
+                    sidebarPath: require.resolve('./sidebars.js'),
+                    routeBasePath: '/',
+                    // Please change this to your repo.
+                    editUrl: 'https://github.com/flagsmith/flagsmith-docs/tree/main',
+                    lastVersion: 'current',
```
Comment (khvn26) [inline docs/docusaurus.config.js]: We'll need to change this, right?
```suggestion
                    editUrl: 'https://github.com/flagsmith/flagsmith/tree/main/docs/',
                    lastVersion: 'current',
```

File: .github/workflows/docs-pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: Docs Lint/Link Checks
+
+on:
+  push:
+    branches:
+      - main
```
Comment (matthewelwell) [inline .github/workflows/docs-pull-request.yml]: This workflow yaml file is called `docs-pull-request.yml` but we're running it on pushes to the main branch? Something needs to change here.

File: .github/workflows/docs-pull-request.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: 'npm'
+          cache-dependency-path: 'docs/package-lock.json'
+
+      - name: npm install
+        run: npm ci --include=dev --ignore-scripts
+      - name: Run docusuarus build to check for broken links etc
+        if: github.event_name == 'pull_request'
```
Comment (matthewelwell) [inline .github/workflows/docs-pull-request.yml]: Related to the above, I think this means that the workflow won't actually do anything other than `npm ci` on pushes to main.

**Review by matthewelwell — CHANGES_REQUESTED** [review]

File: .github/workflows/docs-pull-request.yml (hunk 3 above)
Comment (dabeeeenster) [reply .github/workflows/docs-pull-request.yml]: Updated

File: .github/workflows/docs-pull-request.yml (hunk 2 above)
Comment (dabeeeenster) [reply .github/workflows/docs-pull-request.yml]: Updated

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**